src/lib/config.ts

```diff
@@ -46,6 +46,13 @@ export const requiredVar = (varName: string): string => {
 	return s;
 };
 
+export const optionalVar = <R>(
+	varName: string,
+	defaultValue: R,
+): string | R => {
```

Comment:
There's no point making this generic -- it can always return `string`. If the value is supposed to be an integer, you'll have to parseInt anyway.

Reply:
My point was just to avoid repeating `process.env.`, which made prettier to add line breaks and made the code less consistent and harder to quickly scan. Will revert.

Reply:
@dfunckt even though I liked the consistency of not having to access `process.env` directly anywhere in the config vars section, I reverted this as per your comment :disappointed:

Reply:
@thgreasi I didn't mean to remove the function, read my comment again. What I'm saying is that the *function* need not be generic over some `R` -- it could just have the signature:

```
export const optionalVar = (varName: string, defaultValue: string): string
```

Reply:
Oooooh, my bad :sweat_smile: 
@dfunckt isn't there a case that we indeed want an envvar to have `undefined` as a default value instead of an empty string?

Reply:
There is, but what's the function you wrote can't be used for it -- what's the point of providing a default if undefined is permitted? You get zero benefit out of the function.

If you literally just want to avoid having to write process.env, just drop the `defaultValue` argument and type the return value as `string | undefined`.

Reply:
Updated @dfunckt with func call override, so that it's `string | undefined` when no default value is provided, or `string` when one is.